packages/shell-api/src/stream-processor.ts

```diff
@@ -57,6 +58,37 @@ export default class StreamProcessor extends ShellApiWithMongoClass {
     });
   }
 
+  /**
+   * modify can be used like so:
+   *  Change the pipeline:
+   *    sp.name.modify(pipeline)
+   *  Change the pipeline with additional options:
+   *    sp.name.modify(pipeline, {resumeFromCheckpoint: false})
+   *  For modify requests that don't change the pipeline:
+   *    sp.name.modify({resumeFromCheckpoint: false})
+   */
+  @returnsPromise
+  async modify(
+    pipelineOrOptions: Document[] | Document,
+    options: Document = {}
```

Comment:
The method parameters were defined in the [Scope](https://docs.google.com/document/d/1wQ_e2TrJKegyei0nDFajVejE6nqahmftieyZ3UEMnXE/edit?tab=t.0) and [PD](https://docs.google.com/document/d/1Qvw4fI6DrSxqJbhODnHaY1p3aJIz0632KsJ-qK401d8/edit?tab=t.0#heading=h.wdpdgdpcthgg). I think [watch](https://www.mongodb.com/docs/manual/reference/method/db.collection.watch/) has a similar structure that support arguments of different types. But if this is against the best mongosh practices we can modify the scope.

Reply:
This is fine, but:
- Please add TS overloads that describe the different signatures, i.e. `modify(options: Document): ...` and `modify(pipeline: Document[], options: Document): ...`
- Please add return types to the definitions

Reply:
Done, thanks!